realm/realm-library/src/objectServer/java/io/realm/SyncCredentials.java

```diff
@@ -86,72 +112,56 @@
-        }
-        return new SyncCredentials(IdentityProvider.FACEBOOK, facebookToken, null);
-    }
-
-    /**
-     * Creates credentials based on a Google login.
+     * Creates credentials based on a login with username and password. These credentials will only be verified
+     * by the Object Server.  The user is not created if she does not exist.
      *
-     * @param googleToken a google userIdentifier acquired by logging into Google.
+     * @param username username of the user.
+     * @param password the users password.
      * @return a set of credentials that can be used to log into the Object Server using
-     *         {@link SyncUser#loginAsync(SyncCredentials, String, SyncUser.Callback)}
+     *         {@link SyncUser#loginAsync(SyncCredentials, String, SyncUser.Callback)}.
      * @throws IllegalArgumentException if user name is either {@code null} or empty.
      */
-    public static SyncCredentials google(String googleToken) {
-        if (googleToken == null || googleToken.equals("")) {
-            throw new IllegalArgumentException("Non-null 'googleToken' required.");
-        }
-        return new SyncCredentials(IdentityProvider.GOOGLE, googleToken, null);
+    public static SyncCredentials usernamePassword(String username, String password) {
+        return usernamePassword(username, password, false);
     }
 
     /**
      * Creates a custom set of credentials. The behaviour will depend on the type of {@code identityProvider} and
      * {@code userInfo} used.
      *
-     * @param identityProvider provider used to verify the credentials.
      * @param userIdentifier String identifying the user. Usually a username of userIdentifier.
+     * @param identityProvider provider used to verify the credentials.
      * @param userInfo data describing the user further or {@code null} if the user does not have any extra data. The
      *              data will be serialized to JSON, so all values must be mappable to a valid JSON data type. Custom
      *              classes will be converted using {@code toString()}.
      * @return a set of credentials that can be used to log into the Object Server using
      *         {@link SyncUser#loginAsync(SyncCredentials, String, SyncUser.Callback)}.
      * @throws IllegalArgumentException if any parameter is either {@code null} or empty.
      */
-    public static SyncCredentials custom(String identityProvider, String userIdentifier, Map<String, Object> userInfo) {
-        if (identityProvider == null || identityProvider.equals("")) {
-            throw new IllegalArgumentException("Non-null 'identityProvider' required.");
-        }
-        if (userIdentifier == null || userIdentifier.equals("")) {
-            throw new IllegalArgumentException("Non-null 'userIdentifier' required.");
-        }
+    public static SyncCredentials custom(String userIdentifier, String identityProvider, Map<String, Object> userInfo) {
```

Comment:
We need to add a changelog item to the Object Server section about this, as it is a breaking semantic change.